### PR TITLE
stat/distuv: more accurate calculation of Normal.CDF and LogNormal.CDF

### LINKED ISSUE #577: stat/distuv: normal CDF can be more accurate at negative x-values
### What are you trying to do?

CDF of normal distribution is not accurate enough at large negative x-values.

### What did you do?

```golang
package main

import (
	"fmt"
	"gonum.org/v1/gonum/stat/distuv"
)

func main() {
	cdf := distuv.Normal{Sigma:1}.CDF
	x1, x2 := search(-1000, 1000, cdf)
	fmt.Printf("\t CDF(%.20f) == %.20e \n", x1, cdf(x1))
	fmt.Printf("\t CDF(%.20f) == %.20e \n", x2, cdf(x2))
}

func search(startX, rangeX float64, f func(float64) float64) (x1, x2 float64) {
	y0 := f(startX)
	for {
		endX := startX + rangeX
		rangeX *= 0.5
		mid := startX + rangeX
		if startX == mid || endX == mid {
			return startX, endX
		}
		midValue := f(mid)
		if midValue == y0 {
			startX = mid
		}
	}
}
```

**output:**
```
         CDF(-8.37438892306745685801) == 0.00000000000000000000e+00
         CDF(-8.37438892306745508165) == 5.55111512312578270212e-17
```

Accuracy of CDF can be improved with implementation bellow:

```golang
func (n Normal) CDF(x float64) float64 {
	return 0.5 * math.Erfc(-(x-n.Mu)/(n.Sigma*math.Sqrt2))
}
```

Current implementation:
```golang
func (n Normal) CDF(x float64) float64 {
	return 0.5 * (1 + math.Erf((x-n.Mu)/(n.Sigma*math.Sqrt2)))
}
```

**output from modified CDF:**
```
         CDF(-38.47536573040455465389) == 0.00000000000000000000e+00
         CDF(-38.47536573040454754846) == 4.94065645841246544177e-324
```

### PR DESCRIPTION
**Pull request by bigflood**
Fixes #577

Please take a look.

### CODE REVIEW

File: stat/distuv/lognormal_test.go [hunk 1]
```diff
@@ -42,3 +42,15 @@ func TestLognormal(t *testing.T) {
 		checkProbQuantContinuous(t, i, x, f, tol)
 	}
 }
+
+func TestLognormalCDFPrecision(t *testing.T) {
```
Comment (btracey) [inline stat/distuv/lognormal_test.go]: Could you rename to `TestLognormalIssue577`?

File: stat/distuv/norm_test.go [hunk 2]
```diff
@@ -169,3 +169,15 @@ func BenchmarkNormalQuantile(b *testing.B) {
 		}
 	}
 }
+
+func TestNormalCDFPrecision(t *testing.T) {
```
Comment (btracey) [inline stat/distuv/norm_test.go]: Could you rename to`TestNormalIssue577`

**Comment by btracey** [comment]
Thanks for the contribution!

File: stat/distuv/lognormal_test.go (hunk 1 above)
Comment (kortschak) [reply stat/distuv/lognormal_test.go]: And please add a link to the issue.

File: stat/distuv/norm_test.go (hunk 2 above)
Comment (kortschak) [reply stat/distuv/norm_test.go]: Also add a link here.

File: stat/distuv/lognormal_test.go [hunk 3]
```diff
@@ -42,3 +42,15 @@ func TestLognormal(t *testing.T) {
 		checkProbQuantContinuous(t, i, x, f, tol)
 	}
 }
+
+func TestLognormalIssue577(t *testing.T) {
```
Comment (kortschak) [inline stat/distuv/lognormal_test.go]: Please add a doc comment linking to the issue:
```
// See https://github.com/gonum/gonum/issues/577 for details.
```

File: stat/distuv/norm_test.go [hunk 4]
```diff
@@ -169,3 +169,15 @@ func BenchmarkNormalQuantile(b *testing.B) {
 		}
 	}
 }
+
+func TestNormalIssue577(t *testing.T) {
```
Comment (kortschak) [inline stat/distuv/norm_test.go]: And here.

**Comment by btracey** [comment]
Thanks for the change.

**Pull request merged**